mkdocs/docs/configuration.md

```diff
@@ -222,6 +222,12 @@ catalog:
     uri: sqlite:////tmp/pyiceberg.db
 ```
 
+| Key           | Example                                                      | Default | Description                                                                                                                                                                                    |
+| ------------- | ------------------------------------------------------------ | ------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| uri           | postgresql+psycopg2://username:password@localhost/mydatabase |         | SQLAlchemy backend URL for the catalog database (see [documentation for URL format](https://docs.sqlalchemy.org/en/20/core/engines.html#backend-specific-urls))                                |
+| echo          | true                                                         | false   | SQLAlchemy engine [echo param](https://docs.sqlalchemy.org/en/20/core/engines.html#sqlalchemy.create_engine.params.echo) to log all statements to the default log handler                      |
+| pool_pre_ping | true                                                         | false   | SQLAlchemy engine [pool_pre_ping param](https://docs.sqlalchemy.org/en/20/core/engines.html#sqlalchemy.create_engine.params.pool_pre_ping) to test connections for liveness upon each checkout |
```

Comment:
I have one more suggestion! Almost all of the configuration options are using dashes (`-`), instead of underscores (`_`). Should we follow that convention here as well?

Reply:
I understand your point of view, and I'm not against it.  But here is mine for what it is worth 😜 

I voluntarily kept the same name as SQLAlchemy thinking that eventually, SQLCatalog would support a pass-through to set the kwargs of `create_engine()`.  The first time, it was `echo`, this time is `pool_pre_ping`, will there be another case in a couple of weeks where we'll need another PR to fix a new issue needing another arg to be set?  

So if the plan is to eventually adopt such a pass-through, I would vote for using the same name as SQLAlchemy, even though it goes against pyiceberg's convention.  
On the opposite, if the idea is to always have pyiceberg be in control of the supplied properties, then it makes sense to enforce the convention and rename this property.

Reply:
Okay, fair argument. I can also be convinced otherwise.

Let me give a counterargument: The catalog should not be tied to a specific implementation. By leaking implementation details through the public API (something that we're already doing by using the same naming) we tie it to SqlAlchemy.

Let's ask @kevinjqliu to be the tiebreaker here.

Reply:
> By leaking implementation details through the public API (something that we're already doing by using the same naming) we tie it to SqlAlchemy.

From a public API standpoint, I agree with this statement. 
For the public API of `SQL catalog`, we should keep the same convention. 
But it looks like this particular table is part of the `SQLite` section, so I think it's fine to have SqlAlchemy-specific key/value here.

Reply:
Thanks, let's move this forward 🚀 Thanks @cccs-eric for the PR, and thanks @kevinjqliu for the review